test/k4FWCoreTest/src/components/ExampleFunctionalTransformerHist.cpp

```diff
@@ -36,14 +50,25 @@ struct ExampleFunctionalTransformerHist final
   // This is the function that will be called to produce the data
   edm4hep::MCParticleCollection operator()(const edm4hep::MCParticleCollection& input) const override {
     // Fill the histogram with the energy of one particle
-    ++m_histograms[input[0 + !m_firstParticle.value()].getEnergy()];
+    ++m_histogram[input[0 + !m_firstParticle.value()].getEnergy()];
+#if GAUDI_MAJOR_VERSION >= 39
+    ++m_customHistogram[input[0 + !m_firstParticle.value()].getEnergy()];
+#endif
     // Return an empty collection since we don't care about the collection
     return {};
   }
 
 private:
   // This is the histogram that will be filled, 1 is the number of dimensions of the histogram (1D)
-  mutable Gaudi::Accumulators::RootHistogram<1> m_histograms{this, "Histogram Name", "Histogram Title", {100, 0, 10.}};
+  mutable Gaudi::Accumulators::StaticRootHistogram<1> m_histogram{
+      this, "Histogram Name", "Histogram Title", {100, 0, 10.}};
+
+public:
+#if GAUDI_MAJOR_VERSION >= 39
+  // This is a histogram with title, name and bins that can be set from python
+  void registerCallBack(Gaudi::StateMachine::Transition, std::function<void()>) {}
```

Comment:
I suppose this is necessary for any algorithm that has histograms? In that case we need to document it.

Reply:
This is necessary for histograms with configurable properties, I added a comment about that.